### PR TITLE
AppVeyor: Python 2.7, 3.6, 3.8, and 3.10

### LINKED ISSUE #132: pytest tests/x86_64/test_load.py is failing
% `pytest tests/x86_64/test_load.py`
```
================================== test session starts ==================================
platform darwin -- Python 3.10.6, pytest-7.1.3, pluggy-1.0.0
rootdir: /Users/cclauss/Python/itinerant_futurizer/PeachPy
collected 1 item

tests/x86_64/test_load.py F                                                       [100%]

======================================= FAILURES ========================================
____________________________________ LoadAsm.runTest ____________________________________

self = <tests.x86_64.test_load.LoadAsm testMethod=runTest>

    def runTest(self):

        x = Argument(int32_t)
        y = Argument(float_)

        with Function("Multiply", (x, y), double_) as asm_multiply:
            reg_x = GeneralPurposeRegister32()
            LOAD.ARGUMENT(reg_x, x)

            xmm_y = XMMRegister()
            LOAD.ARGUMENT(xmm_y, y)

            xmm_x = XMMRegister()
            CVTSI2SD(xmm_x, reg_x)
            CVTSS2SD(xmm_y, xmm_y)

            MULSD(xmm_x, xmm_y)

            RETURN(xmm_x)

>       py_multiply = asm_multiply.finalize(abi.detect()).encode().load()

tests/x86_64/test_load.py:28:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <peachpy.x86_64.function.Function object at 0x103d3ef20>, abi = None

    def finalize(self, abi):
        from peachpy.x86_64.abi import ABI
        if not isinstance(abi, ABI):
>           raise TypeError("%s is not an ABI object" % str(abi))
E           TypeError: None is not an ABI object

peachpy/x86_64/function.py:308: TypeError
================================ short test summary info ================================
FAILED tests/x86_64/test_load.py::LoadAsm::runTest - TypeError: None is not an ABI object
=================================== 1 failed in 0.07s ===================================
```

### PR DESCRIPTION
**Pull request by cclauss**
Closes #98
Closes #132

https://www.appveyor.com/docs/lang/python

Replace nosetest with pytest.  For [pytest discovery](https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html), `git mv test tests`

Failing tests in #131 ~and #132~

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
   matrix:
   - PYTHON: C:\Python27-x64
   - PYTHON: C:\Python36-x64
   - PYTHON: C:\Python37-x64
+  - PYTHON: C:\Python38-x64
+  - PYTHON: C:\Python39-x64
+  - PYTHON: C:\Python310-x64
+  # - PYTHON: C:\Python311-x64  # On or after 05 Oct 2022
 install:
-- '"%PYTHON%\Scripts\pip" install -r test\requirements.txt'
```
Comment (Maratyszcza) [inline appveyor.yml:8]: I prefer to use `test/requirements.txt`, this ensures that dependencies are correctly specified. Edit `requirements.txt` if needed.

File: appveyor.yml [hunk 2]
```diff
@@ -1,12 +1,19 @@
...
+- '"%PYTHON%\Scripts\pip" install pytest wheel'
 build_script:
-- '"%PYTHON%\Scripts\pip" install .'
+- '"%PYTHON%\Scripts\pip" install --editable .'
 test_script:
-- '"%PYTHON%\Scripts\nosetests"'
+- '"%PYTHON%\Scripts\pytest" test --ignore=test/arm/test_arm.py --ignore=test/x86_64/test_load.py'
```
Comment (Maratyszcza) [inline appveyor.yml:19]: Why is `test_load` skipped?

**Review by Maratyszcza — CHANGES_REQUESTED** [review]

File: appveyor.yml (hunk 2 above)
Comment (cclauss) [reply appveyor.yml]:
```
FAILED tests/arm/test_arm.py::TestARM::runTest - NameError: name 'ABI' is not defined
FAILED tests/x86_64/test_load.py::LoadAsm::runTest - TypeError: None is not an ABI object
```

File: appveyor.yml (hunk 1 above)
Comment (cclauss) [reply appveyor.yml:8]: Done.

File: appveyor.yml (hunk 2 above)
Comment (cclauss) [reply appveyor.yml]: Fixed.  See #132

**Comment by cclauss** [comment]
@Maratyszcza Was there more required on this one?

**Review by Maratyszcza — APPROVED** [review]

**Comment by Maratyszcza** [comment]
Thank you, merged!

**Pull request merged**